src/coreComponents/constitutive/solid/porosity/BiotPorosity.hpp

```diff
@@ -69,25 +69,32 @@ class BiotPorosityUpdates : public PorosityBaseUpdates
   real64 dGrainDensity_dPressure() const { return 1.0 / m_grainBulkModulus; }
 
   GEOSX_HOST_DEVICE
-  void updateFromPressureAndStrain( localIndex const k,
-                                    localIndex const q,
-                                    real64 const & deltaPressure,
-                                    real64 const (&strainIncrement)[6],
-                                    real64 & dPorosity_dPressure,
-                                    real64 & dPorosity_dVolStrain ) const
+  void updateFromPressureTemperatureAndStrain( localIndex const k,
+                                               localIndex const q,
+                                               real64 const & deltaPressure,
+                                               real64 const & deltaTemperature,
+                                               real64 const (&strainIncrement)[6],
+                                               real64 const & thermalExpansionCoefficient,
+                                               real64 & dPorosity_dVolStrain,
+                                               real64 & dPorosity_dPressure,
+                                               real64 & dPorosity_dTemperature ) const
   {
     real64 const biotSkeletonModulusInverse = (m_biotCoefficient[k] - m_referencePorosity[k]) / m_grainBulkModulus;
+    real64 const porosityThermalExpansion = 3 * thermalExpansionCoefficient * m_biotCoefficient[k];
 
-    real64 const porosity = m_porosity_n[k][q] +
-                            +m_biotCoefficient[k] * LvArray::tensorOps::symTrace< 3 >( strainIncrement ) + biotSkeletonModulusInverse * deltaPressure;
-
-    dPorosity_dPressure = biotSkeletonModulusInverse;
+    real64 const porosity = m_porosity_n[k][q]
+                            + m_biotCoefficient[k] * LvArray::tensorOps::symTrace< 3 >( strainIncrement )
+                            + biotSkeletonModulusInverse * deltaPressure
+                            + porosityThermalExpansion * deltaTemperature;
```

Comment:
The sign of this term might be negative because in an isochore drained condition (zero strain and zero pressure change), an increase in temperature would cause a decrease in porosity?

Reply:
The sign convention, definition of `porosityThermalExpansion` and `thermalExpansionCoefficient` needs to be revisited. I would suggest to update the [rst file for Biot porosity](https://github.com/GEOSX/GEOSX/blob/develop/src/coreComponents/constitutive/docs/BiotPorosity.rst) and add adequate refence for the thermal part, which can help to verify the implementation.

Reply:
Yes, thanks for catching the sign issue, it is fixed now. For now it does not matter because the thermal solver will be added in the next PR (here `deltaTemperature` is still always zero), but still. Also thanks for reminding me about the documentation, I will update it in #2195.